modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/minmaxscaling/MinMaxScalerPreprocessor.java

```diff
@@ -71,8 +71,13 @@ public MinMaxScalerPreprocessor(double[] min, double[] max, IgniteBiFunction<K,
         assert res.size() == min.length;
         assert res.size() == max.length;
 
-        for (int i = 0; i < res.size(); i++)
-            res.set(i, (res.get(i) - min[i]) / (max[i] - min[i]));
+        for (int i = 0; i < res.size(); i++) {
+            double scaled = (res.get(i) - min[i]) / (max[i] - min[i]);
+            if (Double.isNaN(scaled))
+                res.set(i, 0);
```

Comment:
Such behavior can break logic of min-max scaler. In my opinion we should write like this:
double denom = max[i] - min[i]
(res.get(i) - min[i]) / (denom > EPS ? denom : 1.0);

Reply:
Disagree. The min max scaler is a function that maps on [0, 1] interval. In you example the result value is not necessary in this interval. Regarding choosing the exact value from this interval, I reproduced sklearn behavior.

Reply:
Okay, new version keeps information when it's possible.